File: Main Project/Backend Spring boot/HomeAppliancesSpring/src/main/java/com/example/demo/entities/Location.java
package com.example.demo.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name="location")
public class Location {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int loc_id;
	@Column
	String address;
	@Column
	String pincode;
	@Column
	String city;
	
	@OneToOne
	@JoinColumn(name="user_id")
	@JsonIgnoreProperties("loc")
	User user;

	public Location(String address, String pincode, String city, User user) {
		super();
		this.address = address;
		this.pincode = pincode;
		this.city = city;
		this.user = user;
	}
	
	

}

File: Main Project/Backend .net/RentIT/Models/Location.cs
using System;
using System.Collections.Generic;

namespace RentIT.Models
{
    public partial class Location
    {
        public int LocId { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? Pincode { get; set; }
        public int? UserId { get; set; }

        public virtual User? User { get; set; }
    }
}
